### PR TITLE
Add all missing analyses config fields

### PR DESCRIPTION
**Pull request by JosephMarinier**
## Description:

Add to the config page all the fields that were missing from the syntax, similarity, and behavioral testing sections.

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

**Comment by nandhinibsn** [comment]
True. this is simpler. I think we can apply these changes.

File: webapp/src/pages/Settings.tsx [hunk 1]
```diff
@@ -551,6 +557,7 @@ const Settings: React.FC = () => {
       {displayToggleSectionTitle("similarity", "Similarity")}
       {getAnalysesCustomization("similarity")}
       {displayToggleSectionTitle("behavioral_testing", "Perturbation Testing")}
+      {getAnalysesCustomization("behavioral_testing")}
```
Comment (nandhinibsn) [inline webapp/src/pages/Settings.tsx:560]: Do we not hide the section if behavioral testing is null? Do we display default config values if it's null?

File: webapp/src/pages/Settings.tsx [hunk 2]
```diff
@@ -551,6 +557,7 @@ const Settings: React.FC = () => {
       {displayToggleSectionTitle("similarity", "Similarity")}
       {getAnalysesCustomization("similarity")}
       {displayToggleSectionTitle("behavioral_testing", "Perturbation Testing")}
```
Comment (nandhinibsn) [inline webapp/src/pages/Settings.tsx:559]: Display name should be changed to "Behavioral Testing"

Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx:559]: Good catch! I'll change it in this PR. Thank you!

File: webapp/src/pages/Settings.tsx (hunk 1 above)
Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx:560]: Right, I was waiting for your input [here](https://github.com/ServiceNow/azimuth/pull/421/files#r1099033775) before changing that. What do you think?

File: webapp/src/pages/Settings.tsx [hunk 3]
```diff
@@ -93,20 +93,20 @@ const KeyValuePairs: React.FC = ({ children }) => (
 );
 
 const displayKeywordArguments = (name: string, kwargs: Record<string, any>) => (
-  <Box display="grid">
+  <Box key={name} display="flex" flexDirection="column">
```
Comment (JosephMarinier) [inline webapp/src/pages/Settings.tsx:96]: @nandhinibsn
> Changing it to flex won't shrink the content when we scroll?

With the `display="grid"`, I was getting this:
image

With `display="flex" flexDirection="column"`, I get:
image

Can you reproduce any issue with `flex`?

File: webapp/src/pages/Settings.tsx (hunk 1 above)
Comment (nandhinibsn) [reply webapp/src/pages/Settings.tsx:560]: Okay, I'm so sorry I did not respond there. Yes, I do prefer hiding the content; but we disable in similarity and postprocessor.

File: webapp/src/pages/Settings.tsx (hunk 2 above)
Comment (nandhinibsn) [reply webapp/src/pages/Settings.tsx:559]: Thank you so much!

File: webapp/src/pages/Settings.tsx (hunk 1 above)
Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx:560]: No worries!
Right, showing the default values, but grayed out, is also interesting. You have done it for similarity and postprocessor fields, except it doesn't work for read-only fields.
image
We could fix that. Or we can hide. Maybe we can ask the team in standup tomorrow?

File: webapp/src/pages/Settings.tsx (hunk 3 above)
Comment (nandhinibsn) [reply webapp/src/pages/Settings.tsx:96]: Hey @JosephMarinier, I tried later with adding flex, and it did not shrink. So, it should be fine. Thanks.

File: webapp/src/pages/Settings.tsx (hunk 1 above)
Comment (nandhinibsn) [reply webapp/src/pages/Settings.tsx:560]: Yes, sure, we can ask for clarification. True, disable feature is not added to the read-only part.

File: webapp/src/pages/Settings.tsx (hunk 3 above)
Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx:96]: Awesome, thank you!

**Pull request merged**